status/flowcells.py

```diff
@@ -416,10 +416,19 @@ def get(self, query):
             self.write(t.generate(gs_globals=self.application.gs_globals, user=self.get_current_user(), readsdata=ordereddata, query=query))
         else:
             xfc_view = self.application.x_flowcells_db.view("samples/lane_clusters", reduce=False)
+            bioinfo_view = self.application.bioinfo_db.view("latest_data/sample_id")
             for row in xfc_view[query:"{}Z".format(query)]:
                 if not row.key in data:
                     data[row.key]=[]
                 data[row.key].append(row.value)
+            #To check if sample is failed on lane level
+            for row in bioinfo_view:
+                if row.key[3] in data:
+                    for k,v in data.items():
+                        if k == row.key[3]:
+                            for fcl in v:
+                                if row.key[1] + ':' + row.key[2] == fcl['fcp']:
+                                    fcl['sample_status'] = row.value['sample_status']
```

Comment:
If you don't want to write a new view, then you can do this
```suggestion
            for row in bioinfo_view[[query, None, None, None]:[f'{query}Z', 'ZZ', 'ZZ', 'ZZ']]:
                    if row.key[3] in data:
                        for fcl in data[row.key[3]]:
                            if row.key[1] + ':' + row.key[2] == fcl['fcp']:
                                fcl['sample_status'] = row.value['sample_status']
```

Reply:
I don't fully understand it, but it's a lot faster! 👏 Thank you! 😄